libraries/AP_Param/AP_Param.cpp

```diff
@@ -257,6 +257,21 @@ bool AP_Param::duplicate_key(uint16_t vindex, uint16_t key)
     return false;
 }
 
+// check for duplicate group ids
+bool AP_Param::duplicate_id(const struct GroupInfo* group_info)
+{
+    uint8_t type;
```

Comment:
Why is there this "type" variable? Assigning it in this Boolean check is nasty.

Reply:
Done.Thanks !